core/src/main/scala/chisel3/Printable.scala

```diff
@@ -377,7 +378,7 @@ case object Percent extends Printable {
 }
 
 /** Represents the hierarchical name in the Verilog (`%m`) */
-case object HierarchicalName extends Printable {
+case object HierarchicalModuleName extends Printable {
```

Comment:
Consider a trait called `SpecialSubstitution` which has an abstract member called "substitutionString" or something. That can then simplify the code above and avoid the need for introducing the wrapping `{{`...`}}` as that is automatic.

This can all be done as soon as we add another substitution, e.g., `SimulationTime` 😉 .

Reply:
Will do as part of SimulationTime which should come today or tomorrow.